galaxyproject/galaxy
# enabling Reset button for Rule Builder tab

This is a fix for #13405; instead of disabling the Reset button with the flag enableReset (as the other tabs in the Upload modal use), I've set it with a computed function that returns true if anything has been put in the sourceContent text box.

The other modals get enableReset flipped in a method `_updateStateForCounters` in the UploadBoxMixin, but when I tried investigating that, I found that that method is never called by RulesInput, and when I tried to force it to call that method, the variables used to flip enableReset are undefined for this type of upload. 

Therefore, this is the most clear-cut solution to enabling the reset button. 

![Screenshot from 2022-02-24 13-57-17](https://user-images.githubusercontent.com/26912553/155589327-7d65f80a-5122-493d-853f-1ed2c96be8dc.png)


I also put in a quick fix for the tooltips on the upload modal buttons. They were all displaying the same thing, "Download from web or upload from disk" which should only apply to the headers. I added tooltips that match the same content as the button itself, which makes them not very helpful tooltips, but is better than something that is completely off base from what they actually do. I'm willing to modify the text and make them more beneficial tooltips, but there's only so many ways to say what something like "Close" or "Start" will do. 
Optimally, these don't have a tooltip at all. Undoubtedly that was the original intention. I'm not sure why the title for the modal-header from `UploadModal.vue` factored down to all of these buttons. Any suggestions? 


## How to test the changes?
(Select all options that apply)
  1. Type something in the Rule-based upload modal, and see that the reset button is enabled.
  2. Click the reset button and see the source content clear away.
  3. Note that the reset button is disabled again.

## License
